source/Halibut/Transport/PollingClient.cs

```diff
@@ -95,5 +117,56 @@ void ExecutePollingLoop(object ignored)
                 }
             }
         }
+
+        async Task ExecutePollingLoopAsyncCatchingExceptions()
+        {
+            try
```

Comment:
I suspect that this try/catch becomes redundant if we implement the token in the way I described before.

Reply:
Noted that it is a "just in case" and to avoid unobserved exceptions